homeassistant/components/homematicip_cloud.py

```diff
@@ -5,143 +5,181 @@
 https://home-assistant.io/components/homematicip_cloud/
 """
 
+import asyncio
 import logging
-from socket import timeout
-
 import voluptuous as vol
 
-from homeassistant.core import callback
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.dispatcher import (dispatcher_send,
-                                              async_dispatcher_connect)
-from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.discovery import async_load_platform
 from homeassistant.helpers.entity import Entity
 
-REQUIREMENTS = ['homematicip==0.8']
+REQUIREMENTS = ['homematicip==0.9.1']
 
 _LOGGER = logging.getLogger(__name__)
 
 DOMAIN = 'homematicip_cloud'
 
+COMPONENTS = [
+    'sensor'
+]
+
 CONF_NAME = 'name'
 CONF_ACCESSPOINT = 'accesspoint'
 CONF_AUTHTOKEN = 'authtoken'
 
 CONFIG_SCHEMA = vol.Schema({
-    vol.Optional(DOMAIN): [vol.Schema({
-        vol.Optional(CONF_NAME, default=''): cv.string,
+    vol.Optional(DOMAIN, default=[]): vol.All(cv.ensure_list, [vol.Schema({
+        vol.Optional(CONF_NAME): vol.Any(None, cv.string),
```

Comment:
Don't allow `None`.

Reply:
Not specifying a name will have the same result.

Reply:
Then I need you expertise, so far I have two access points in my config, like:

`homematicip_cloud:`
`  - name:`
`   accesspoint: !secret homematicip_accesspoint`
`    authtoken: !secret homematicip_authtoken`
`  - name: loc2`
`    accesspoint: !secret homematicip_accesspoint-loc2`
`    authtoken: !secret homematicip_authtoken-loc2`

First one without a prefix for the components, second one with HUB2.

Reply:
We should never have users specifying a key without a value. The following config should give you the same result.
```yaml
homematicip_cloud:
  - accesspoint: !secret homematicip_accesspoint
    authtoken: !secret homematicip_authtoken
  - name: loc2
    accesspoint: !secret homematicip_accesspoint-loc2
    authtoken: !secret homematicip_authtoken-loc2
```